app/models/portfolio.rb

```diff
@@ -1,7 +1,7 @@
 class Portfolio < ApplicationRecord
   belongs_to :user
-  has_many :portfolio_transactions
-  has_many :portfolio_stocks
+  has_many :portfolio_transactions, dependent: :destroy
+  has_many :portfolio_stocks, dependent: :destroy
```

Comment:
This may also be a candidate for `restrict_with_error` or `restrict_with_exception`